TouchPortalAPI/__init__.py

```diff
@@ -251,29 +252,46 @@ def removeStateMany(self, states:list):
     def choiceUpdate(self, choiceId:str, values:list):
         if choiceId:
             if isinstance(values, list):
-                self.send({"type": "choiceUpdate", "id": choiceId, "value": values})
+                self.__send({"type": "choiceUpdate", "id": choiceId, "value": values})
             else:
                 raise TypeError(f'choiceUpdate() values argument needs to be a list not a {type(values)}')
 
     def choiceUpdateSpecific(self, stateId:str, values:list, instanceId:str):
         if stateId and instanceId:
             if isinstance(values, list):
-                self.send({"type": "choiceUpdate", "id": stateId, "instanceId": instanceId, "value": values})
+                self.__send({"type": "choiceUpdate", "id": stateId, "instanceId": instanceId, "value": values})
             else:
                 raise TypeError(f'choiceUpdateSpecific() values argument needs to be a list not a {type(values)}')
 
     def settingUpdate(self, settingName:str, settingValue):
         if settingName and settingName not in self.currentSettings or self.currentSettings[settingName] != settingValue:
-            self.send({"type": "settingUpdate", "name": settingName, "value": settingValue})
+            self.__send({"type": "settingUpdate", "name": settingName, "value": settingValue})
             self.currentSettings[settingName] = settingValue
 
+    def showNotification(self, notificationId:str, title:str, msg:str, options):
+        if notificationId and title and msg and options and options:
+            if isinstance(options, list):
+                jsonOptions = []
+                for option in options:
+                    if isinstance(option, dict):
+                        if option["id"] and option["title"]:
+                            jsonOptions.insert(option)
```

Comment:
It's also pointless to copy the data into a new array if you just want to validate the whole `options` array. Just validate `options` and then use that directly, or throw the error if it doesn't validate. Copying isn't "free" and in this case just a waste of memory and cpu cycles.